### ISSUE TITLE
Shader that uses TIME does not force editor to continuously update in Godot4

### ISSUE THREAD

**Issue by t3nk3y** [godotengine/godot#51957]
### Godot version

v4.0.dev.custom_build [e599f1bdf]

### System information

Ubuntu 21.04

### Issue description

When using a shader that uses `TIME` to provide animation, the editor should continuously update the viewport as it did Godot 3, however in Godot 4, it is not doing so.

Doing anything that refreshes the viewport, such as scrolling or zooming will update the shader.

This can be worked around by enabling "Update Continuously" in the Editor settings.

Here is how it should work:
![Peek 2021-08-21 14-16](https://user-images.githubusercontent.com/26716328/130331353-0e3f2b1e-67e9-4f0e-8c5c-bd6ab7e4b0ae.gif)

Here is how it DOES work:
![Peek 2021-08-21 14-15](https://user-images.githubusercontent.com/26716328/130331362-61f36245-68e9-459d-bf86-ce634fd681a3.gif)

### Steps to reproduce

Place any object in the 3D editor that has a ShaderMaterial that includes `TIME` to provide animation.

NOTE: Be sure "Update Continuously" is off in the Editor settings, as this will work around the issue.

### Minimal reproduction project

This project has an example.  Just load the project and the main scene should immediately fail to work right.(so long as update continuously is not checked in editor settings)
[ShaderEditorIssueExample.zip](https://github.com/godotengine/godot/files/7025980/ShaderEditorIssueExample.zip)

If anyone like the little shader I put in here, feel free to use it, no credits needed.

**Comment by Zireael07** [comment]
I believe this has been turned off for performance reasons?

**Comment by Calinou** [comment]
> I believe this has been turned off for performance reasons?

This was likely not done intentionally, as this is not the behavior in `3.x`. A proper way to fix this would be https://github.com/godotengine/godot/pull/43011, as this is something that needs to be consciously enabled by the user.

**Comment by t3nk3y** [comment]
I agree that #43011 would be a sensible fix.  Feel free to close this issue if you want.

**Comment by Calinou** [comment]
I'll leave this open as this warrants further discussion as to whether this behavior should be kept.

**Comment by clayjohn** [comment]
We need to make a ``redraw_request`` when a material has the ``uses_time`` property set to ``true``. Here is how it is done in 2D

https://github.com/godotengine/godot/blob/5f69218edc4f4af574c7a65f077e2e0bf1f28365/servers/rendering/renderer_rd/renderer_canvas_render_rd.cpp#L1372-L1374
https://github.com/godotengine/godot/blob/5f69218edc4f4af574c7a65f077e2e0bf1f28365/servers/rendering/renderer_rd/renderer_canvas_render_rd.cpp#L1472-L1474

and in sky shaders:
https://github.com/godotengine/godot/blob/036b7a09851fb0bfe97f1c5086ef99b1cfb87d2f/servers/rendering/renderer_rd/renderer_scene_sky_rd.cpp#L1095-L1099

**Comment by jinyoung-lim** [comment]
I'm new to open source and would love to take a stab at this issue.
